### PR TITLE
DCS-1573 Add date of hearing downloads

### PR DESCRIPTION
**Pull request by davidthomas22**
*Add download option page
*Refactor and update testing

### CODE REVIEW

File: integration-tests/integration/downloadReports/videoLinkHearingEvents.cy.js [hunk 1]
```diff
@@ -17,19 +18,26 @@ context('A user can download video link booking events as CSV files', () => {
     cy.task('deleteFolder', downloadsFolder)
   })
 
+  it('selects download by booking option', () => {
+    cy.visit('/video-link-booking-events')
+    const page = DownloadOptionPage.verifyOnPage()
+    const form = page.form()
+    form.dateHearing().type('radio').first().check()
+    page.continueButton().click()
+  })
```
Comment (andrewrlee) [inline integration-tests/integration/downloadReports/videoLinkHearingEvents.cy.js:27]: HearingPage.verifyOnPage()
page.form().heading().contains('Download by hearing date')

File: backend/routes/downloadReports/indexController.test.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
+import { mockRequest, mockResponse } from '../__test/requestTestUtils'
+import IndexController from './indexController'
+
+jest.mock('../../api/whereaboutsApi')
```
Comment (GurnankCheema) [inline backend/routes/downloadReports/indexController.test.ts:4]: don't think we need line 4 as not using mock in the test

**Review by GurnankCheema — DISMISSED** [review]
LGTM

**Review by GurnankCheema — APPROVED** [review]
LGTM

**Pull request merged**